utils/compare_ds.py

```diff
@@ -0,0 +1,274 @@
+                "'%s' to '%s'." % (rule_id, old_oval_def_id, new_oval_def_id)
+            )
+        if show_diffs:
+            try:
+                old_oval_def_doc = old_oval_defs[old_oval_file_name]
+            except KeyError:
+                print(
+                    "Rule '%s' points to '%s' which isn't a part of the "
+                    "old datastream" % (rule_id, old_oval_file_name))
+                return
+            try:
+                new_oval_def_doc = new_oval_defs[new_oval_file_name]
+            except KeyError:
+                print(
+                    "Rule '%s' points to '%s' which isn't a part of the "
+                    "new datastream" % (rule_id, new_oval_file_name))
+                return
+            compare_oval_definitions(
+                old_oval_def_doc, old_oval_def_id, new_oval_def_doc,
+                new_oval_def_id)
+
+
+def compare_remediation_contents(old_r, new_r):
+    if old_r != new_r:
+        diff = "".join(difflib.unified_diff(
+            old_r.splitlines(keepends=True), new_r.splitlines(keepends=True),
+            fromfile="old datastream", tofile="new datastream"))
+        return diff
+    return None
+
+
+def compare_remediations(old_rule, new_rule, remediation_type, show_diffs):
+    system = remediation_type_to_uri[remediation_type]
+    old_fix = old_rule.find("xccdf:fix[@system='%s']" % (system), ns)
+    new_fix = new_rule.find("xccdf:fix[@system='%s']" % (system), ns)
+    rule_id = old_rule.get("id")
+    if (old_fix is None and new_fix is not None):
+        print("New datastream adds %s remediation for rule '%s'." % (
+            remediation_type, rule_id))
+    elif (old_fix is not None and new_fix is None):
+        print("New datastream is missing %s remediation for rule '%s'." % (
+            remediation_type, rule_id))
+    elif (old_fix is not None and new_fix is not None):
+        old_fix_id = old_fix.get("id")
+        new_fix_id = new_fix.get("id")
+        if old_fix_id != new_fix_id:
+            print(
+                "%s remediation ID for rule '%s' has changed from "
+                "'%s' to '%s'." % (
+                    remediation_type, rule_id, old_fix_id, new_fix_id)
+            )
+        if show_diffs:
+            diff = compare_remediation_contents(old_fix.text, new_fix.text)
+            if diff:
+                print("%s remediation for rule '%s' differs:\n%s" % (
+                    remediation_type, rule_id, diff))
+
+
+def process_benchmarks(
+        old_benchmark, new_benchmark, old_oval_defs, new_oval_defs,
+        rule_id, show_diffs):
+    missing_rules = []
+    if rule_id:
+        if not rule_id.startswith(ssg.constants.OSCAP_RULE):
+            rule_id = ssg.constants.OSCAP_RULE + rule_id
+        rules_in_old_benchmark = old_benchmark.findall(
+            ".//xccdf:Rule[@id='%s']" % (rule_id), ns)
+    else:
+        rules_in_old_benchmark = old_benchmark.findall(".//xccdf:Rule", ns)
```

Comment:
Ideally, split this into "get_rules_to_compare" and "compare_rules" functions - the original one is quite big and function names act as comments pointing towards what's going on.